Firestore/core/src/remote/serializer.cc

```diff
@@ -1436,23 +1437,29 @@ std::unique_ptr<WatchChange> Serializer::DecodeExistenceFilterWatchChange(
 
 ExistenceFilter Serializer::DecodeExistenceFilter(
     const google_firestore_v1_ExistenceFilter& filter) const {
-  // Create bloom filter if there is an unchanged_names present in the filter
-  // and inputs are valid, otherwise keep it null.
-  absl::optional<BloomFilter> bloom_filter;
   if (filter.has_unchanged_names) {
-    // TODO(Mila): None of the ported spec tests here actually has the bloom
-    // filter json string, so hard code an empty bloom filter for now. The
-    // actual parsing code will be written in the next PR, where we can validate
-    // the parsing result.
-    // TODO(Mila): handle bloom filter creation failure.
+    pb_bytes_array_t* bitmap_ptr = filter.unchanged_names.has_bits
```

Comment:
Yes, you do need to handle the case where `filter.unchanged_names.has_bits== false`: if it is false then use an empty array instead. The way the code is currently written will crash if `filter.unchanged_names.has_bits== false` because it sets `bitmap_ptr` to `nullptr` and then immediately after `bitmap_ptr->bytes` and `bitmap_ptr->size` are accessed on that null pointer.

I'd recommend something like this:

```cpp
std::vector<uint8_t> bitmap;
if (bitmap_ptr) {
  bitmap = std::vector<uint8_t>(bitmap_ptr->bytes, bitmap_ptr->bytes + bitmap_ptr->size);
}
```